File: elevators-ui/src/app/_model/elevator-dto.ts
import { TransportDto } from "./transport-dto";

export interface ElevatorDto {
    currentLoad: number,
    personsInside: number,
    currentFloor: number,
    nextFloor: number,
    currentDestination: number,
    isTaken: boolean,
    currentTransports: Array<TransportDto>
}

File: elevators-service/src/main/java/com/example/elevatorsservice/elevator/dto/ElevatorDto.java
package com.example.elevatorsservice.elevator.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.List;


@AllArgsConstructor
@Getter
@Setter
public class ElevatorDto {
	private Integer currentLoad;
	private Integer personsInside;
	private Integer currentFloor;
	private Integer nextFloor;
	private Integer currentDestination;
	private boolean isTaken;
	private List<TransportDto> currentTransports;
}
